source/applications/advanced/gamma_correction.py

```diff
@@ -0,0 +1,100 @@
+"""
+Capture a 2D image and apply a configurable gamma correction.
+"""
+
+import argparse
+from pathlib import Path
+import numpy as np
+import cv2
+import zivid
+
+
+def _options():
+    """Configure argparse.
+
+    Returns:
+        argparser arguments
+
+    """
+    parser = argparse.ArgumentParser(
+        description="Capture 2D image and apply gamma correction\
+            \nExample:\n\t $ python gamma_correction.py 2",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    parser.add_argument(
+        "gamma", nargs="?", type=float, help="Gamma correction value", default="2"
+    )
+
+    return parser.parse_args()
+
+
+def adjust_gamma(image, gamma=1.0):
+    """Adjust gamma on image.
+
+    Args:
+        image: OpenCV image to be corrected
+        gamma: Gamma setting to apply
+
+    Returns:
+        Gamma correct (OpenCV) image
+
+    """
+    # build a lookup table mapping the pixel values [0, 255] to
+    # their adjusted gamma values
+    inv_gamma = 1.0 / gamma
+    table = np.array(
+        [((i / 255.0) ** inv_gamma) * 255 for i in np.arange(0, 256)]
+    ).astype("uint8")
+    # apply gamma correction using the lookup table
+    return cv2.LUT(image, table)
+
+
+def _capture_2d_image(camera):
+    """Capture from file, and extract 2D image.
+
+    Args:
+        camera: Zivid Camera handle
+
+    Returns:
+        RGBA image as numpy array
+
+    """
+    settings_2d = zivid.Settings2D()
+    with camera.capture_2d(settings_2d) as frame_2d:
+        image = frame_2d.image()
+        image.save("original.png")
+        return image.to_array()
+
+
+def _main():
+    app = zivid.Application()
+
+    user_options = _options()
+
+    if not app.cameras():
+        print(f"No camera connected, will use file camera instead")
+        camera = app.create_file_camera(Path(__file__).parents[2] / "MiscObjects.zdf")
+    else:
+        camera = app.connect_camera()
+
+    image = _capture_2d_image(camera)
+    original = np.dstack([image["b"], image["g"], image["r"]])
+
+    print(f"Applying gamma correction: {user_options.gamma}")
+    adjusted = adjust_gamma(original, gamma=user_options.gamma)
```

Comment:
Accordingly to the previous comment, this would be bgr_adjusted. It is good to emphesize the weird openCV order b-g-r.